crates/accelerate/src/sparse_observable.rs

```diff
@@ -998,6 +1206,133 @@ impl SparseObservable {
         slf.borrow().eq(&other.borrow())
     }
 
+    fn __add__(&self, other: &Bound<PyAny>) -> PyResult<Py<PyAny>> {
+        let py = other.py();
+        let Some(other) = coerce_to_observable(other)? else {
+            return Ok(py.NotImplemented());
+        };
+        let other = other.borrow();
+        self.check_equal_qubits(&other)?;
+        Ok((self + &other).into_py(py))
+    }
+    fn __radd__(&self, other: &Bound<PyAny>) -> PyResult<Py<PyAny>> {
+        let py = other.py();
+        let Some(other) = coerce_to_observable(other)? else {
+            return Ok(py.NotImplemented());
+        };
+        let other = other.borrow();
+        self.check_equal_qubits(&other)?;
+        Ok((<&SparseObservable as ::std::ops::Add>::add(&other, self)).into_py(py))
+    }
+    fn __iadd__(slf_: Bound<SparseObservable>, other: &Bound<PyAny>) -> PyResult<()> {
+        if slf_.is(other) {
+            *slf_.borrow_mut() *= Complex64::new(2.0, 0.0);
```

Comment:
The reason to have the special case is because you can't borrow `self` as mutable (to hold the result) _and_ immutable (to be the addand).  So there's no easy way to pass it to any existing `add` implementation here in the module, and I used this trick to avoid introducing a clone in the operation.

Reply:
ahh I see, so you would've preferred to have this be the same outcome as `obs + obs` but for performance we just do the multiplication -- but I'm also wondering whether we should then do the same for `obs + obs -> 2 * obs`? 🤔

Reply:
I suppose yeah, it's technically a performance optimisation, though really I think that rewriting it as `2. * obs` was just the thing that came more naturally to me, because I avoid `clone` just by default whenever possible.

I can make the behaviour of the not-in-place forms the same as the in-place forms if you prefer?

Reply:
Yeah I think that would be nice, unless you have concerns/other expectations 🙂